tests/integration/nm/vlan_test.py

```diff
@@ -50,6 +46,7 @@ def _vlan_interface(state):
 
 
 def _get_vlan_current_state(vlan_desired_state):
+    nm.nmclient.client(refresh=True)
```

Comment:
IMHO it would be better to do this in the get_device_by_name() function but since it segfaults there, this is an acceptable workaround. Please add a FIXME, though, so we remove this eventually.

Reply:
It cannot be done in the production NM package code because there we are already operating inside the mainloop (for apply at least). I guess that is the cause of the segfaults.

Regarding the FIXME, I do not think we can "fix" it. In the NM integration tests we run the low level NM api, and create the mainloop ourself in the test. This last assert checks if the interface has been removed (those action have been executed in the mainloop), and it does not need a mainloop, just a client refresh.
Bottom line, there is no fix here unless you can think of something else we can do to make this nicer.

Reply:
Keeping the mainloop running would ensure that the client will be refreshed AFAIU, so it would be a fix... But I see that it would be difficult to do this in the test code as well.